drivers/bluetooth/hci/hci_nxp.c

```diff
@@ -331,44 +460,51 @@ static int bt_nxp_open(const struct device *dev, bt_hci_recv_t recv)
 int bt_nxp_setup(const struct device *dev, const struct bt_hci_setup_params *params)
 {
 	ARG_UNUSED(dev);
-	ARG_UNUSED(params);
 
-	int ret;
+	int ret = 0;
 
 	do {
-#if CONFIG_HCI_NXP_SET_CAL_DATA
-		ret = bt_nxp_set_calibration_data();
-		if (ret < 0) {
-			LOG_ERR("Failed to set calibration data");
-			break;
-		}
-#if CONFIG_HCI_NXP_SET_CAL_DATA_ANNEX100
-		/* After send annex55 to CPU2, CPU2 need reset,
-		 * a delay of at least 20ms is required to continue sending annex100
-		 */
-		k_sleep(Z_TIMEOUT_MS(20));
-
-		ret = bt_nxp_set_calibration_data_annex100();
-		if (ret < 0) {
-			LOG_ERR("Failed to set calibration data");
-			break;
+		if (IS_ENABLED(CONFIG_HCI_NXP_SET_CAL_DATA)) {
+			ret = bt_nxp_set_calibration_data();
+			if (ret < 0) {
+				LOG_ERR("Failed to set calibration data");
+				break;
+			}
+			if (IS_ENABLED(CONFIG_HCI_NXP_SET_CAL_DATA_ANNEX100)) {
+				/* After send annex55 to CPU2, CPU2 need reset,
+				 * a delay of at least 20ms is required to continue sending annex100
+				 */
+				k_sleep(Z_TIMEOUT_MS(20));
+
+				ret = bt_nxp_set_calibration_data_annex100();
+				if (ret < 0) {
+					LOG_ERR("Failed to set calibration data");
+					break;
+				}
+			}
 		}
-#endif /* CONFIG_HCI_NXP_SET_CAL_DATA_ANNEX100 */
-#endif /* CONFIG_HCI_NXP_SET_CAL_DATA */
 
-#if CONFIG_HCI_NXP_ENABLE_AUTO_SLEEP
-		ret = nxp_bt_set_host_sleep_config();
-		if (ret < 0) {
-			LOG_ERR("Failed to set host sleep config");
-			break;
+		if (IS_ENABLED(CONFIG_HCI_NXP_ENABLE_AUTO_SLEEP)) {
+			ret = nxp_bt_set_host_sleep_config();
+			if (ret < 0) {
+				LOG_ERR("Failed to set host sleep config");
+				break;
+			}
+
+			ret = nxp_bt_enable_controller_autosleep();
+			if (ret < 0) {
+				LOG_ERR("Failed to configure controller autosleep");
+				break;
+			}
 		}
 
-		ret = nxp_bt_enable_controller_autosleep();
-		if (ret < 0) {
-			LOG_ERR("Failed to configure controller autosleep");
-			break;
+		if (IS_ENABLED(CONFIG_BT_HCI_SET_PUBLIC_ADDR)) {
+			ret = bt_nxp_set_mac_address(&(params->public_addr));
```

Comment:
You can't use `if (IS_ENABLED(CONFIG_FOO))` if the function that's called is behind `#ifdef`. You have three options:

1. Remove the #ifdef from the static function
2. Use #ifdef here as well instead of `IS_ENABLED()`
3. Make public address setting always enabled for the driver by selecting it in Kconfig.

If 3 is acceptable I'd go for that, since then you simplify the driver code noticeably. Right now it's clear that you haven't even tested disabling the `BT_HCI_SET_PUBLIC_ADDR` option, so I have a hunch that this should be ok.

Reply:
the function is stubbed if `BT_HCI_SET_PUBLIC_ADDR` is not enabled, see:
https://github.com/zephyrproject-rtos/zephyr/pull/79432/files#diff-bdedfa33c7fab1d576fa6ea8ea8f9e1d8651ffeb0861ea146fdeaa7eb8a3b99fR78

So the case where the option is disabled is covered by this. And actually, we tested it because this driver is used by another platform (RW612) which doesn't support this option, so it's disabled on this platform.

Is it ok for you if we let it as it is ?

Reply:
> the function is stubbed if `BT_HCI_SET_PUBLIC_ADDR` is not enabled

Ah, I missed that part. In that case it's probably fine (as long as you fix the `#if` vs `#if defined()` stuff)